### PR TITLE
Harden reservation and loan lifecycle edge cases

### PR DESCRIPTION
**Pull request by fabiodalez-dev**
## Summary
- harden reservation promotion and capacity fairness by pausing suspended, expired-card, and at-limit patrons without losing FIFO identity
- expire promoted reservation loans that were never approved, and recheck borrower eligibility at scheduled activation and physical pickup
- make immediate approval notifications deadline-aware and add truthful awaiting-approval and lost/damaged email flows
- persist terminal circulation emails in a leased outbox with hourly and daily retry plus exponential backoff
- support assessed lost/damaged charges, archived-title barcode returns, and stronger queue/date database guards
- bump schema to 0.7.80 with fresh-install and upgrade support

## Testing
- 157 supported standalone PHP unit test files passed
- circulation hardening: 22 checks
- email outbox behavior: 8 checks
- loan edge cases: 68 checks
- expiry sweep behavior: 34 checks
- PHPStan passed in the commit hook
- composer validate --no-check-publish
- npm run test:ci-policy
- locale parity and soft-delete policy checks
- PHP lint and git diff --check

## Not run
- loan-reservation-real-world-25, loan-review-0764, and multiple-copy-loans-238 require an explicitly configured dedicated E2E_DB_NAME because they perform global maintenance and cleanup

## Summary by CodeRabbit

* **Nuove funzionalità**
  * Gestione dei rientri per copie perse o danneggiate, con importo addebitato opzionale.
  * Nuove notifiche email per esiti dei prestiti e prenotazioni in attesa di approvazione.
  * Recupero automatico delle email non consegnate tramite coda persistente.
  * Controlli su idoneità, tessere scadute e limite di prestiti attivi.
  * Scadenza automatica delle prenotazioni non approvate.

* **Correzioni**
  * Migliorata la gestione e il riordino delle code di prenotazione.
  * Impedito il ritiro da parte di utenti non idonei.

* **Stile**
  * Aggiornati layout responsive, intestazioni e pulsanti dell’Emeroteca.

* **Localizzazione**
  * Aggiunte traduzioni e template email in più lingue.

### CODE REVIEW

File: app/Controllers/PrestitiController.php [hunk 1]
```diff
@@ -1367,6 +1368,20 @@ public function processReturn(Request $request, Response $response, mysqli $db,
...
+        // decimal comma from Italian keyboards, but reject negative, malformed
+        // or oversized values rather than silently coercing them.
+        $penaltyRaw = str_replace(',', '.', trim((string) ($data['sanzione'] ?? '0')));
+        if (!in_array($nuovo_stato, ['perso', 'danneggiato'], true)) {
+            $penaltyRaw = '0';
+        } elseif ($penaltyRaw === '') {
+            $penaltyRaw = '0';
+        }
+        if (preg_match('/^(?:0|[1-9]\d{0,7})(?:\.\d{1,2})?$/D', $penaltyRaw) !== 1) {
+            return $response->withHeader('Location', url('/admin/loans/returned/' . $id) . '?error=invalid_penalty')->withStatus(302);
```
[the remark this answers is not in the document]
Comment (fabiodalez-dev) [reply app/Controllers/PrestitiController.php:1380]: Keeping url('/admin/loans/returned/…') as-is: admin routes are English literals by project decision (issue #145) — route_path()/RouteTranslator cover only the localized user-facing routes, admin paths never go through the i18n system. The same pattern is used two lines above for the invalid_status redirect and across every admin controller.

File: app/Support/mail_templates/da_DK.php [hunk 2]
```diff
@@ -281,6 +281,14 @@
...
+    ],
+    'reservation_awaiting_approval' => [
+        'subject' => '📚 Eksemplar tildelt — afvent afhentningsbekræftelse',
+        'body' => '<h2>Et eksemplar er tildelt din reservation</h2><p>Hej {{utente_nome}},</p><p>Din reservation står nu først i køen, og et eksemplar er afsat til {{data_inizio}} - {{data_fine}}.</p><p><strong>Bog:</strong> {{libro_titolo}} — {{libro_autore}} ({{libro_isbn}})</p><div style="background-color:#fef3c7;padding:15px;border-left:4px solid #f59e0b"><p><strong>Kom ikke på biblioteket endnu.</strong></p><p>Anmodningen skal stadig godkendes. Du får en ny e-mail med afhentningsfristen, når bogen er klar.</p></div><p><a href="{{book_url}}">Se bogen</a> · <a href="{{profile_url}}">Mine lån</a></p>',
+    ],
```
[the remark this answers is not in the document]
Comment (fabiodalez-dev) [reply app/Support/mail_templates/da_DK.php:291]: Fixed in 9f50dbac — and the gap was wider than da_DK: ALL five locale seeds were missing both templates. Each seed now carries loan_copy_outcome and reservation_awaiting_approval with the exact subject/body of its mail_templates/<locale>.php (base IT from SettingsMailTemplates), ids appended after the last row, INSERT/INSERT IGNORE style matched per file. Every row proven to import into a sandbox clone of email_templates.

File: app/Support/NotificationService.php [hunk 3]
```diff
@@ -1941,6 +1955,69 @@ public function sendLoanReturnedNotification(int $loanId): bool {
...
+                    false,
+                    $locale
+                ),
+                'note' => trim((string) ($loan['note'] ?? '')) ?: ($noNotes[$language] ?? $noNotes['it']),
+                'sanzione' => number_format(
+                    $amount,
+                    2,
+                    in_array(substr($locale, 0, 2), ['it', 'de', 'fr', 'da'], true) ? ',' : '.',
+                    ''
+                ) . ' €',
```
[the remark this answers is not in the document]
Comment (fabiodalez-dev) [reply app/Support/NotificationService.php]: Fixed in 9f50dbac: the amount now uses ConfigStore::get('app.currency', 'EUR') — € symbol for EUR, the ISO code otherwise — keeping the locale-aware separators.

File: app/Support/NotificationService.php [hunk 4]
```diff
@@ -2070,9 +2198,125 @@ private function sendWithRetry(string $email, string $template, array $variables
...
+            $error = is_array($variables) ? 'delivery failed' : 'invalid variables_json';
+            $release = $this->db->prepare("
+                UPDATE email_delivery_outbox
+                SET attempts = ?, available_at = ?, claim_token = NULL,
+                    claimed_at = NULL, last_error = ?
+                WHERE id = ? AND claim_token = ?
+            ");
+            $release->bind_param('issis', $attempts, $retryAt, $error, $id, $token);
+            $release->execute();
+            $release->close();
```
[the remark this answers is not in the document]
Comment (fabiodalez-dev) [reply app/Support/NotificationService.php:2338]: Fixed in 9f50dbac: OUTBOX_MAX_ATTEMPTS = 10 next to the retry constants; past the threshold the row is deleted under the claim token and the discarded delivery is recorded with SecureLogger::error(). Backoff/release behaviour unchanged below the cap.

File: app/Views/prestiti/restituito_prestito.php [hunk 5]
```diff
@@ -177,6 +178,23 @@ class="rounded-lg border-2 border-gray-300 bg-white px-4 py-3 text-gray-900 font
...
+                <label id="sanzione-field" class="hidden flex-col gap-2">
+                    <span class="text-sm font-bold text-gray-900"><?= __("Importo addebitato") ?></span>
+                    <div class="relative">
+                        <span class="pointer-events-none absolute inset-y-0 left-0 flex items-center pl-4 text-gray-500">€</span>
+                        <input
+                            id="sanzione"
+                            name="sanzione"
+                            type="text"
+                            inputmode="decimal"
+                            value="<?= HtmlHelper::e((string) ($prestito['sanzione'] ?? '0.00')); ?>"
```
[the remark this answers is not in the document]
Comment (fabiodalez-dev) [reply app/Views/prestiti/restituito_prestito.php]: Fixed in 9f50dbac: htmlspecialchars((string) …, ENT_QUOTES, 'UTF-8') with the same null fallback.

File: installer/database/triggers.sql [hunk 6]
```diff
@@ -184,6 +196,86 @@ BEGIN
...
+    IF NEW.stato = 'attiva' AND NEW.queue_position IS NOT NULL AND EXISTS (
+        SELECT 1 FROM prenotazioni r
+        WHERE r.libro_id = NEW.libro_id AND r.queue_position = NEW.queue_position
+          AND r.stato = 'attiva'
+    ) THEN
+        SIGNAL SQLSTATE '45000'
+            SET MESSAGE_TEXT = 'Posizione in coda già occupata per questo libro.';
+    END IF;
+END$$
+DELIMITER ;
```
[the remark this answers is not in the document]
Comment (fabiodalez-dev) [reply installer/database/triggers.sql:234]: Fixed in 9f50dbac: all five cancellation flows (UserActionsController, ReservationsAdminController, LoanApprovalController, NcipServerPlugin, DataIntegrity) now do the two-phase move — active non-NULL positions shift to +1000000 temporaries inside the existing transaction, then final positions are assigned — mirroring ReservationManager. Reproduced against the live triggers: the pre-fix reorder (NULL row + position 1) dies with the SIGNAL, the post-fix one succeeds. DataIntegrity's "fixed" counter now compares against the original position so the temp shift doesn't inflate it.

File: locale/de_DE.json [hunk 7]
```diff
@@ -7287,5 +7287,17 @@
...
+  "Inviata quando una prenotazione ottiene una copia e attende l'approvazione del prestito.": "Wird gesendet, wenn einer Vormerkung ein Exemplar zugewiesen wurde und die Ausleihgenehmigung aussteht.",
+  "Copia persa o danneggiata": "Verlorenes oder beschädigtes Exemplar",
+  "Inviata quando un prestito viene chiuso dichiarando la copia persa o danneggiata.": "Wird gesendet, wenn eine Ausleihe mit einem verlorenen oder beschädigten Exemplar geschlossen wird.",
+  "Data di chiusura del prestito": "Abschlussdatum der Ausleihe",
+  "Esito registrato per la copia fisica": "Erfasster Zustand des Exemplars",
+  "Note registrate dall'operatore": "Vom Personal erfasste Notizen",
+  "Importo addebitato per la copia": "Für das Exemplar berechneter Betrag",
+  "Importo non valido: usa un valore positivo con al massimo due decimali.": "Ungültiger Betrag: Verwende einen positiven Wert mit höchstens zwei Dezimalstellen.",
+  "Importo addebitato": "Berechneter Betrag",
+  "Facoltativo; viene registrato e comunicato per copie perse o danneggiate.": "Optional; wird bei verlorenen oder beschädigten Exemplaren erfasst und mitgeteilt."
```
[the remark this answers is not in the document]
Comment (fabiodalez-dev) [reply locale/de_DE.json:7302]: Fixed in 9f50dbac: "…beim Wiederholen der in der Warteschlange befindlichen E-Mails".

File: tests/loan-edge-cases.unit.php [hunk 8]
```diff
@@ -89,6 +90,8 @@ function loadEnv(string $path): array
 // triggers otherwise fall back to the database's UTC CURRENT_DATE(),
 // which disagrees with app.timezone between 22:00 and 24:00 UTC.
 \App\Support\DateHelper::synchronizeDatabaseSession($db);
+$edgeSettings = new SettingsRepository($db);
+$originalMaxActiveLoans = $edgeSettings->get('loans', 'max_active_loans_per_user', '0');
```
[the remark this answers is not in the document]
Comment (fabiodalez-dev) [reply tests/loan-edge-cases.unit.php]: Fixed in 9f50dbac: the original get() no longer passes a default, so NULL preserves the row's absence; both restore points (scenario 66 and the cleanup) call delete() when the row did not exist and set() with the raw original otherwise.

File: tests/loan-expiry-sweeps-behavior.unit.php [hunk 9]
```diff
@@ -173,6 +175,19 @@
...
 
+// ═════════ Fixture C: promoted reservation never approved before window end ═════════
+[$bookC, [$copyC]] = $makeBook('C', 1);
+[$userC] = $makeUser('c');
+$stmt = $db->prepare(
+    "INSERT INTO prestiti (libro_id, copia_id, utente_id, data_prestito, data_scadenza, stato, origine, attivo)
+     VALUES (?, ?, ?, ?, ?, 'pendente', 'prenotazione', 0)"
+);
+$sC = $d(-10); $eC = $d(-2);
+$stmt->bind_param('iiiss', $bookC, $copyC, $userC, $sC, $eC);
```
[the remark this answers is not in the document]
Comment (fabiodalez-dev) [reply tests/loan-expiry-sweeps-behavior.unit.php:191]: Fixed in 9f50dbac: fixture C now starts with the copy in stato 'prenotato' and copie_disponibili = 0, so the release assertion can actually fail when checkExpiredReservations() does not free the copy. Suite green (34/34).

File: app/Support/DataIntegrity.php [hunk 10]
```diff
@@ -1098,26 +1098,44 @@ public function fixDataInconsistencies(): array {
...
+                // si assegnano le posizioni finali — il trigger BEFORE UPDATE
+                // rifiuta i duplicati riga per riga durante il riordino.
+                $shiftStmt = $this->db->prepare("
+                    UPDATE prenotazioni
+                    SET queue_position = queue_position + 1000000
+                    WHERE libro_id = ? AND stato = 'attiva' AND queue_position IS NOT NULL
+                ");
+                $shiftStmt->bind_param('i', $bookId);
+                $shiftStmt->execute();
+                $shiftStmt->close();
```
[the remark this answers is not in the document]
Comment (fabiodalez-dev) [reply app/Support/DataIntegrity.php]: Fixed in 684e1731, in all seven reorder paths (the five cancellation flows AND both pre-existing ReservationManager methods). One refinement over the suggested MAX-only offset: the renumbering paths use max(MAX(active queue_position), N) — plain MAX is insufficient when legacy NULL-position rows push the final count N past the shifted zone (1 row at position 1 plus two NULLs: MAX=1 would shift that row to 2 and collide with the second final assignment). The decrement path in updateQueuePositions keeps MAX alone since its final positions never reach the shifted zone. Reproduced against the live triggers: the adversarial queue (active positions 1 and 1000001) fails with the fixed offset and succeeds with the dynamic one; the multi-NULL edge also succeeds. No literal 1000000 remains.
